### PR TITLE
Bump cdap and hadoop version

### PR DESCRIPTION
**Pull request by psainics**
## Bump cdap and hadoop version ( Unit Test Fixes )

Bump up CDAP (`6.11.0-SNAPSHOT`) and Hadoop (`3.3.6`) version for error management.

### Unit Tests Fixed

- Modified `HttpSourceETLTest.java`
- Modified `HttpStreamingSourceETLTest.java`
- Modified `HTTPSinkTest.java`

### Other changes

1. Exclusion `slf4j-reload4j`
```xml
<exclusion>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-reload4j</artifactId>
</exclusion>
```

- Caused error on unit tests
Screenshot 2024-12-02 at 4 23 36 PM

2. Adding `EXPLORE_ENABLED`
- This constant was missing in the new version.

### Sanity Check on new build

image
image
image
image

### CODE REVIEW

**Comment by psainics** [comment]
@itsankit-google i am not confident on the `exclusion` can you please confirm i have done it correctly.

File: src/test/java/io/cdap/plugin/http/etl/HttpStreamingSourceETLTest.java [hunk 1]
```diff
@@ -56,11 +56,11 @@ public class HttpStreamingSourceETLTest extends HttpSourceETLTest {
   private static final ArtifactSummary APP_ARTIFACT = new ArtifactSummary("data-streams", "1.0.0");
   private static final int WAIT_FOR_RECORDS_TIMEOUT_SECONDS = 60;
   private static final long WAIT_FOR_RECORDS_POLLING_INTERVAL_MS = 100;
+  public static final String EXPLORE_ENABLED = "explore.enabled";
 
   @ClassRule
   public static final TestConfiguration CONFIG =
-    new TestConfiguration(Constants.Explore.EXPLORE_ENABLED, false,
-                          Constants.AppFabric.SPARK_COMPAT, Compat.SPARK_COMPAT);
+    new TestConfiguration(EXPLORE_ENABLED, false, Constants.AppFabric.SPARK_COMPAT, Compat.SPARK_COMPAT);
```
Comment (itsankit-google) [inline src/test/java/io/cdap/plugin/http/etl/HttpStreamingSourceETLTest.java]: Did you try removing this config altogether?

**Review by itsankit-google — COMMENTED** [review]
Can you please add `build.yml` & `build-report.yml` files?

ref: https://github.com/data-integrations/salesforce/tree/develop/.github/workflows

File: src/test/java/io/cdap/plugin/http/etl/HttpStreamingSourceETLTest.java (hunk 1 above)
Comment (psainics) [reply src/test/java/io/cdap/plugin/http/etl/HttpStreamingSourceETLTest.java]: The `HttpStreamingSourceETLTest` test keep falling, i cannot be sure if removing this is correct !
image

File: pom.xml [hunk 2]
```diff
@@ -76,14 +76,14 @@
...
+    <cdap.version>6.11.0-SNAPSHOT</cdap.version>
     <commons.version>3.9</commons.version>
     <common.codec.version>1.12</common.codec.version>
     <gson.version>2.8.5</gson.version>
     <googleauth.version>0.4.0</googleauth.version>
-    <hadoop.version>2.3.0</hadoop.version>
+    <hadoop.version>3.3.6</hadoop.version>
     <httpcomponents.version>4.5.9</httpcomponents.version>
-    <hydrator.version>2.10.0-SNAPSHOT</hydrator.version>
+    <hydrator.version>2.10.0</hydrator.version>
```
Comment (itsankit-google) [inline pom.xml]: bump it to `2.13.0-SNAPSHOT` for consistency

Comment (psainics) [reply pom.xml]: Bumped 👍

File: src/test/java/io/cdap/plugin/http/etl/HttpStreamingSourceETLTest.java (hunk 1 above)
Comment (itsankit-google) [reply src/test/java/io/cdap/plugin/http/etl/HttpStreamingSourceETLTest.java]: `java.lang.NoClassDefFoundError` usually means dependency conflicts needs to be resolved at pom level.

Comment (psainics) [reply src/test/java/io/cdap/plugin/http/etl/HttpStreamingSourceETLTest.java]: Bump to `spark-streaming` and `jackson` version fixes this error !

image

**Comment by psainics** [comment]
The unit tests failing here are also failing in `develop` branch, we can have a separate PR addressing broken unit tests, and merge this one as a bump up so other developers can work with error management task.

**Comment by itsankit-google** [comment]
> The unit tests failing here are also failing in `develop` branch, we can have a separate PR addressing broken unit tests, and merge this one as a bump up so other developers can work with error management task.

Build with unit tests is a required step to pass when we bump any dependencies. We need to make sure it is green before proceeding with any other change.

**Comment by psainics** [comment]
🥳 Unit Tests Fixed

**Review by itsankit-google — APPROVED** [review]

**Pull request merged**